### DISCUSSION TITLE
IPv6 uris in leafnodes remotes configuration

### DISCUSSION THREAD

**Discussion by akaanautomaatio** [nats-io/nats-server#4877] [Q&A] [answered]
Are IPv6 uris supported in leafnode remotes configuration, and if so how should I format the address?

I've tried pretty much every format i can imagine off, with and without dev name (br-lan in that case), with and without brackets... nothing seems to work.

Config file:
```
listen: "[::]:4222"
leafnodes {
    remotes = [
        {
            url: "nats-leaf://[fe80::e65f:1ff:fe4e:cb0c%br-lan]:7422"
        }
    ]
}
```

Results:
```
root@host:/dir# nats-server -l /opt/nats-server.log -c /var/run/nats.conf -DVV
nats-server: /var/run/nats.conf:6:13: error parsing leafnode url ["nats-leaf://[fe80::e65f:1ff:fe4e:cb0c%br-lan]:7422"]
```

**Comment by derekcollison** [comment, marked as answer]
It can parse them but may present issues with routing. 

For instance this is parsed correctly for the demo server, but will fail to route to the host.

```
leafnodes {
  remotes = [
    { url: "nats-leaf://[2604:1380:4643:2d00::1]:7422" }
  ]
}
```

We suggest placing the IPv6 in a DNS record or /etc/hosts entry for best results.
